lib/Conversion/TritonGPUToLLVM/FuncOpToLLVM.cpp

```diff
@@ -73,6 +73,7 @@ struct FuncOpConversion : public ConvertOpToLLVMPattern<triton::FuncOp> {
       amendedInputTy.push_back(sharedPtrTy);
     }
     amendedInputTy.push_back(globalPtrTy);
+    amendedInputTy.push_back(profilePtrTy);
```

Comment:
Would it be possible to elide the extra argument if profiling is not enabled?

Reply:
Since it eventually is also needed down in the Python launcher code in places like here:
https://github.com/triton-lang/triton/blob/main/third_party/nvidia/backend/driver.py#L304
I think the concern is it makes it very easy if the interface ever changes to start writing into actual kernel arg regions. Especially since the global scratch buffer does similar things like assuming it's index is always number of kernel args - 2.